cloud/services/loadbalancers/loadbalancers.go

```diff
@@ -206,5 +215,128 @@ func (s *Service) getFrontendIPConfigs(lbSpec azure.LBSpec) ([]network.FrontendI
+		return []network.OutboundRule{}
+	}
+	return []network.OutboundRule{
+		{
+			Name: to.StringPtr(OutboundNAT),
+			OutboundRulePropertiesFormat: &network.OutboundRulePropertiesFormat{
+				Protocol:                 network.LoadBalancerOutboundRuleProtocolAll,
+				IdleTimeoutInMinutes:     to.Int32Ptr(4),
+				FrontendIPConfigurations: &frontendIDs,
+				BackendAddressPool: &network.SubResource{
+					ID: to.StringPtr(azure.AddressPoolID(s.Scope.SubscriptionID(), s.Scope.ResourceGroup(), lbSpec.Name, lbSpec.BackendPoolName)),
+				},
+			},
+		},
+	}
+}
+
+func (s *Service) getLoadBalancingRules(lbSpec azure.LBSpec, frontendIDs []network.SubResource) []network.LoadBalancingRule {
+	if lbSpec.Role == infrav1.APIServerRole {
+		// We disable outbound SNAT explicitly in the HTTPS LB rule and enable TCP and UDP outbound NAT with an outbound rule.
+		// For more information on Standard LB outbound connections see https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-connections.
+		var frontendIPConfig network.SubResource
+		if len(frontendIDs) != 0 {
+			frontendIPConfig = frontendIDs[0]
+		}
+		return []network.LoadBalancingRule{
+			{
+				Name: to.StringPtr(LBRuleHTTPS),
+				LoadBalancingRulePropertiesFormat: &network.LoadBalancingRulePropertiesFormat{
+					DisableOutboundSnat:     to.BoolPtr(true),
+					Protocol:                network.TransportProtocolTCP,
+					FrontendPort:            to.Int32Ptr(lbSpec.APIServerPort),
+					BackendPort:             to.Int32Ptr(lbSpec.APIServerPort),
+					IdleTimeoutInMinutes:    to.Int32Ptr(4),
+					EnableFloatingIP:        to.BoolPtr(false),
+					LoadDistribution:        network.LoadDistributionDefault,
+					FrontendIPConfiguration: &frontendIPConfig,
+					BackendAddressPool: &network.SubResource{
+						ID: to.StringPtr(azure.AddressPoolID(s.Scope.SubscriptionID(), s.Scope.ResourceGroup(), lbSpec.Name, lbSpec.BackendPoolName)),
+					},
+					Probe: &network.SubResource{
+						ID: to.StringPtr(azure.ProbeID(s.Scope.SubscriptionID(), s.Scope.ResourceGroup(), lbSpec.Name, HTTPSProbe)),
+					},
+				},
+			},
+		}
+	}
+	return []network.LoadBalancingRule{}
+}
+
+func (s *Service) getBackendAddressPools(lbSpec azure.LBSpec) []network.BackendAddressPool {
+	return []network.BackendAddressPool{
+		{
+			Name: to.StringPtr(lbSpec.BackendPoolName),
+		},
+	}
+}
+
+func (s *Service) getProbes(lbSpec azure.LBSpec) []network.Probe {
+	if lbSpec.Role == infrav1.APIServerRole {
+		return []network.Probe{
+			{
+				Name: to.StringPtr(HTTPSProbe),
+				ProbePropertiesFormat: &network.ProbePropertiesFormat{
+					Protocol:          network.ProbeProtocolHTTPS,
+					RequestPath:       to.StringPtr("/healthz"),
+					Port:              to.Int32Ptr(lbSpec.APIServerPort),
+					IntervalInSeconds: to.Int32Ptr(15),
+					NumberOfProbes:    to.Int32Ptr(4),
+				},
+			},
+		}
+	}
+	return []network.Probe{}
+}
+
+func probeExists(probes []network.Probe, probe network.Probe) bool {
```

Comment:
note: these exists helpers look _very_ similar... I tried to combine them into one and use reflect to get the Name field, like this:

```
func exists(existing []interface{}, item interface{}) bool { 
wanted := reflect.ValueOf(&item).Elem().FieldByName("Name").Elem().String() 
for _, v:=range existing {
     name:=reflect.ValueOf(&v).Elem().FieldByName("Name").Elem().String()
     if name == wanted { 
        return true 
    }
} 
return false 
}
```
But then I realized because I have to deal with slices of structs, converting them to `[]interface{}` is costly since it's O(n) so in the end I decided the duplication of code is not worse.